R/mod-review-section.R

```diff
@@ -93,23 +103,31 @@ mod_review_section_server <- function(input, output, session, synapse, syn,
       updateSelectInput(
         session = getDefaultReactiveDomain(),
         "section",
-        choices = c("", synapseforms::get_main_sections(
-          sub_data,
-          input$submission
-        ))
+        choices = c(
+          "",
+          get_sections(
+            sub_data,
+            input$submission
+          )
+        )
       )
     }
   })
 
-  submission <- reactive({ input$submission })
-  section <- reactive({ input$section })
+  submission <- reactive({
+    input$submission
+  })
+  section <- reactive({
+    input$section
+  })
 
   ## Show section
   to_show <- reactive({
-    dplyr::filter(
+    sub_section <- dplyr::filter(
       sub_data,
-      form_data_id == submission() & section == section() &!is.na(sub_data$response)
+      submission == submission() & step == section() & !is.na(sub_data$response)
```

Comment:
```suggestion
      .data$submission == submission() & .data$step == section() & !is.na(sub_data$response)
```
Suggesting this to avoid the R CMD check warning `no visible binding for global variable ‘step’`. We should also put `#' @importFrom rlang .data` in the roxygen comments for this function.

Reply:
That's interesting. I need to look into .data since that seems handy. Thanks!

Reply:
Fixed.